Add functions to get milli/micro/nano-seconds from a DateTime

beneills opened the pull request:
Using the underlying naive::NaiveTime fractional part, we compute
the number of milli/micro/nano-seconds since the last second boundary.

The reason for not computing elapsed time since 1970 is because we
would hit potential issues of i64s not being large enough (the range
would be strictly smaller than the 64bit-timestamp range, causing
compatibility issues).

This was written in response to issue #74.

lifthrasiir commented:
I think it is a fine addition, but I have some questions:

- Shouldn't the names of those methods follow [`Duration::subsec_nanos`](https://doc.rust-lang.org/std/time/struct.Duration.html#method.subsec_nanos)? (My opinion: Yes.)

- How should they behave on the leap seconds? The nanosecond part may exceed 999,999,999 in that case, and you need a clear policy for that. (My opinion: Let they return the value exceeding 1s, and document that caveat in big letters. The proper formatting should go through `format` which already handles leap seconds correctly, and the accessors should be considered as a "raw" format.)

- Probably nitpicking, but please avoid adding spaces in irrelevant lines ;) You can surgery them with `git add -p` or similar tools.

beneills commented:
> Shouldn't the names of those methods follow Duration::subsec_nanos? (My opinion: Yes.)

Renamed according to this schema.

> How should they behave on the leap seconds? The nanosecond part may exceed 999,999,999 in that case, and you need a clear policy for that. (My opinion: Let they return the value exceeding 1s, and signal that caveat loudly. The proper formatting should go through format which already handles leap seconds correctly, and the accessors should be considered as a "raw" format.)

Added prominent Rustdoc warnings to all accessors.

> Probably nitpicking, but please avoid adding spaces in irrelevant lines ;) You can surgery them with git add -p or similar tools.

Fixed.  Apologies for this: was trying out a new editor.

lifthrasiir commented:
@beneills I've squashed and merged the PR. Thank you!
